test/bin/manage_build_cache.sh

```diff
@@ -0,0 +1,169 @@
+#!/bin/bash
+#
+# This script should be run on the build host to manage cache of build artifacts
+set -euo pipefail
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+# shellcheck source=test/bin/common.sh
+source "${SCRIPTDIR}/common.sh"
+
+AWS_BUCKET_NAME="${AWS_BUCKET_NAME:-microshift-build-cache}"
+REL_SUBDIR=
+TAG_SUBDIR=
+
+usage() {
+    cat - <<EOF
+${BASH_SOURCE[0]} (upload|download|exist|latest|keep) [options]
+
+Manage build cache at the '${AWS_BUCKET_NAME}' AWS S3 bucket. The script
+assumes that the bucket exists and it is configured for read-write access
+using 'aws s3 (ls|sync|rm)' operations.
+
+The cache directory structure is '${AWS_BUCKET_NAME}/<release>/<tag>'.
+
+  -h        Show this help.
+
+  upload:   Upload build artifacts from the local disk to the specified
+            '${AWS_BUCKET_NAME}/<release>/<tag>' AWS S3 bucket.
+
+  download: Download build artifacts from the specified
+            '${AWS_BUCKET_NAME}/<release>/<tag>' AWS S3 bucket to
+            the local disk.
+
+  verify:   Exit with 0 status if the specified
+            '${AWS_BUCKET_NAME}/<release>/<tag>' sub-directory exists
+            and contains files, 1 otherwise.
+
+  latest:   Update the '${AWS_BUCKET_NAME}/<release>/latest' directory
+            in the AWS S3 bucket to a copy of the specified '<tag>'.
+
+  keep:     Delete all data from the '${AWS_BUCKET_NAME}/<release>'
+            AWS S3 bucket, only keeping the '<tag>' sub-directory.
+
+Options:
+
+  -r <release>  The release sub-directory in the AWS S3 bucket.
```

Comment:
would it be easier to use branch? `main, 4.14, 4.15` vs `main, release-4.14, release-4.15`? less parsing, just getting from CI ref inputs

Reply:
Yes, it's a good comment. Let me change the name of `release` to `branch` to make it more explicit.